docs/REST API/rest-api.md

```diff
@@ -9138,6 +9141,133 @@ Add App Store (VPP) app purchased in Apple Business Manager.
 
 `Status: 200`
 
+### List Fleet-maintained apps
+
+> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows.
+
+List available Fleet-maintained apps.
+
+`GET /api/v1/fleet/software/fleet_maintained_apps`
+
+#### Parameters
+
+| Name | Type | In | Description |
+| ---- | ---- | -- | ----------- |
+| team_id       | integer | query | **Required**. The team ID. Filters Fleet-maintained apps to only include apps available for the specified team.  |
+
+#### Example
+
+`GET /api/v1/fleet/software/fleet_maintained_apps?team_id=3`
+
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "fleet_maintained_apps": [
+    {
+      "id": "1",
+      "name": "1Password",
+      "version": "8.10.40",
+      "platform": "darwin"
+    },
+    {
+      "id": "2",
+      "name": "Adobe Acrobat Reader",
+      "version": "24.002.21005",
+      "platform": "darwin"
+    },
+    {
+      "id": "3",
+      "name": "Box Drive",
+      "version": "2.39.179",
+      "platform": "darwin"
+    },
+  ],
+  "meta": {
+    "has_next_results": false,
+    "has_previous_results": false
+  }
+}
+```
+
+### Get Fleet-maintained app
+
+> **Experimental feature**. This feature is undergoing rapid improvement, which may result in breaking changes to the API or configuration surface. It is not recommended for use in automated workflows.
+
+Returns information about the specified Fleet-maintained app.
+
+`GET /api/v1/fleet/software/fleet_maintained_apps/:id`
+
+#### Parameters
+
+| Name | Type | In | Description |
+| ---- | ---- | -- | ----------- |
+| id   | integer | path | **Required.** The Fleet-maintained app's ID. |
+
+
+#### Example
+
+`GET /api/v1/fleet/software/fleet_maintained_apps/1`
+
+##### Default response
+
+`Status: 200`
+
+```json
+{
+  "fleet_maintained_app": {
+    "id": 1,
+    "name": "1Password",
+    "file_name": "1Password-8.10.44-aarch64.zip",
```

Comment:
Hey @mna, is there any other way to retrieve `filename`? Could we make it if the URL doesn't end with extensions like: .dmg, .zip, .pkg, etc. then Fleet tries to download a file to get the name? Is that too complex? I think having `release` or `stable` as filename could be confusing.

I agree, we should be consistent and use `filename` without `_`. I'll update PR.

Reply:
>  I added it so users understand what will be added to Fleet. Is it .pkg, .dmg, or something else. They'll see the $INSTALLER_PATH variable in the install script, it will help understand what's behind that variable.

Gotcha, makes sense.

> I'm going to bring this to design review today for expedited drafting.

Sounds good, not a blocker for the moment but of course the sooner we can clarify, the better. Thanks!

Reply:
@mna How will Fleet know what should be a install script before adding Fleet-maintained app to Fleet?

At this point. I guess we need to know that file type is `.dmg` in order to have script that mount disk image, moves `.app` from it and unmount.
<img width="847" alt="image" src="https://github.com/user-attachments/assets/5e596f8e-bb5a-484d-8fd2-326cd7d4dabc">

Reply:
@marko-lisica correct, yes we (will) have a JSON file that describes the list of maintained apps, (e.g. it contains the "brew identifier" - the part to use in the API call e.g. `1password` in `https://formulae.brew.sh/api/cask/1password.json`), the bundle identifier associated with this app (since we can't get it from brew) and the "installer format" which is an arbitrary string that defines how the installer is bundled, and comes from your research in the google doc (e.g. "dmg:app"). At least that's the plan at the moment, see https://github.com/fleetdm/fleet/blob/fa62b4c57c83fde6396063e89d7b6f85965b682c/server/mdm/maintainedapps/apps.json#L1.

Maybe we could add the filename as part of this hard-coded JSON file? Only thing is if we want to include the version as part of the filename, that part is dynamic (we could make the filename some kind of template string that we fill with brew data). But that's a possibility!

Reply:
> Could we simplify that, so filename is optional in this apps.json, and we provide it only for apps that have "weird" URLs (WhatsApp and VS Code)?

Absolutely! Makes sense to me.

> I think if we want accurate name we'll need it dynamic (template string that we fill with brew data)

Allright, so we want the version in there and the extension to reflect its format, correct?

Reply:
@mna We decided to cut `filename` and use `name` instead. See Figma [here](https://www.figma.com/design/bA6Y0QSQ3yvZpTAXbYa5Dn/%2318865-Fleet-maintained-apps-for-macOS?node-id=552-6860&node-type=INSTANCE&t=t1pzBh9ozyIFfb4q-11).

Reply:
@marko-lisica sounds good, thanks!